// @inrupt/lit-generated-vocab-inrupt-common/GeneratedVocab/INRUPT_COMMON.js
const { LitVocabTerm, getLocalStore } = require("@inrupt/lit-vocab-term");
const dataFactory = require("@rdfjs/data-model");

const { namedNode } = dataFactory;

/**
  Generated by artifact generator [@inrupt/lit-artifact-generator], version [0.10.13]
  as part of artifact: [vocab-inrupt-common], version: [0.3.9]
  at 'Thursday, May 28, 2020 1:43 AM'.

  Vocabulary built from vocab list file: [/home/pmcb55/Work/Projects/LIT/lit-vocab/inrupt-rdf-vocab/Common/Vocab-List-Inrupt-Common.yml].

  The Inrupt common vocabulary - just commonly used
 terms across all Inrupt products and services.
 */

function _NS(localName) {
  return namedNode(`https://inrupt.com/vocab/common#${localName}`);
}

const INRUPT_COMMON = {
  PREFIX: "inrupt_common",
  NAMESPACE: "https://inrupt.com/vocab/common#",
  PREFIX_AND_NAMESPACE: { "inrupt_common": "https://inrupt.com/vocab/common#" },
  NS: _NS,



  // *******************
  // All the Literals.
  // *******************

  /**
   * Failed to process incoming message as RDF
   */
  errFailedToProcessIncomingRdf: new LitVocabTerm(
    _NS("errFailedToProcessIncomingRdf"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addMessage(`Failed to process incoming message as RDF`, "en"),

  /**
   * Incoming request validated as RDF (it has [{{0}}] triples!)
   */
  msgRequestValidatedAsRdf: new LitVocabTerm(
    _NS("msgRequestValidatedAsRdf"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addMessage(`Incoming request validated as RDF (it has [{{0}}] triples!)`, "en"),
  //
  // Marker allowing us put commas at the end of all lines above (only the last line does not have a comma).
  //
  END_OF_VOCAB: 'End of vocab.'
}

module.exports = INRUPT_COMMON;
