// @inrupt/lit-generated-vocab-inrupt-common/GeneratedVocab/UI_COMMON.js
const { LitVocabTerm, getLocalStore } = require("@inrupt/lit-vocab-term");
const dataFactory = require("@rdfjs/data-model");

const { namedNode } = dataFactory;

/**
  Generated by artifact generator [@inrupt/lit-artifact-generator], version [0.10.13]
  as part of artifact: [vocab-inrupt-common], version: [0.3.9]
  at 'Thursday, May 28, 2020 1:43 AM'.

  Vocabulary built from vocab list file: [/home/pmcb55/Work/Projects/LIT/lit-vocab/inrupt-rdf-vocab/Common/Vocab-List-Inrupt-Common.yml].

  The Inrupt User Interface (UI) common vocabulary -
 commonly used terms across all Inrupt UI components.
 */

function _NS(localName) {
  return namedNode(`https://inrupt.com/vocab/ui/common#${localName}`);
}

const UI_COMMON = {
  PREFIX: "ui_common",
  NAMESPACE: "https://inrupt.com/vocab/ui/common#",
  PREFIX_AND_NAMESPACE: { "ui_common": "https://inrupt.com/vocab/ui/common#" },
  NS: _NS,



  // *******************
  // All the Literals.
  // *******************

  /**
   * Tip: To fix this problem, try and convert to a supported media type (see: <Link> for currently supported types).
   */
  errUnsupportedMediaType: new LitVocabTerm(
    _NS("errUnsupportedMediaType"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addComment(`Tip: To fix this problem, try and convert to a supported media type (see: <Link> for currently supported types).`, "en")
    .addMessage(`Unsupported media type: [{{0}}]`, "en")
    .addMessage(`Tipo de soporte no soportado: [{{0}}]`, "es")
    .addMessage(`Type de support non supporté: [{{0}}]`, "fr"),

  /**
   * Tip: To fix this problem, try and reduce the file size (e.g. crop an image, compress an audio file, etc.).
   */
  errFileUpload_exceededSizeLimit: new LitVocabTerm(
    _NS("errFileUpload_exceededSizeLimit"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addComment(`Tip: To fix this problem, try and reduce the file size (e.g. crop an image, compress an audio file, etc.).`, "en")
    .addMessage(`File [{{0}}] has size [{{1}}] that exceeds the allowable limit of [{{2}}]`, "en")
    .addMessage(`El tamaño [{{1}}] del archivo [{{0}}] excede el límite permitido [{{2}}]`, "es")
    .addMessage(`La taille du fichier [{{1}}], de [{{0}}], dépasse la limite autorisée [{{2}}]`, "fr"),

  /**
   * Tip: To fix this problem, simply select fewer files to upload.
   */
  errFileUpload_exceededMaximumFiles: new LitVocabTerm(
    _NS("errFileUpload_exceededMaximumFiles"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addComment(`Tip: To fix this problem, simply select fewer files to upload.`, "en")
    .addMessage(`Sorry, you have exceeded the maximum number of files allowed per upload: [{{0}}]`, "en")
    .addMessage(`Lo sentimos, has excedido la cantidad máxima de archivos permitidos por carga: [{{0}}]`, "es")
    .addMessage(`Désolé, vous avez dépassé le nombre maximal de fichiers autorisés par téléchargement: [{{0}}]`, "fr"),
  //
  // Marker allowing us put commas at the end of all lines above (only the last line does not have a comma).
  //
  END_OF_VOCAB: 'End of vocab.'
}

module.exports = UI_COMMON;
